### DISCUSSION TITLE
Error State::$name

### DISCUSSION THREAD

**Discussion by mtamagusku** [spatie/laravel-model-states#167] [Q&A]
Hi, i have a issue when deploy in my server. when i want acces to static name attribute of my state class i got this error:
{ "message": "syntax error, unexpected 'Model' (T_STRING), expecting function (T_FUNCTION) or const (T_CONST)", "exception": "ParseError", "file": "/var/www/virtual/<domain>/html/vendor/spatie/laravel-model-states/src/State.php", "line": 17.

But when i try in my local dev, its work.
Some ideas?
Thx

**Comment by brendt** [comment]
I don't know without seeing your code.

**Reply by mtamagusku** [reply]
abstract
![image](https://user-images.githubusercontent.com/68010870/115547837-2036b500-a27d-11eb-8ad8-3723a6c7c6b7.png)

state
![image](https://user-images.githubusercontent.com/68010870/115548538-f7fb8600-a27d-11eb-9ff1-81a2517c1bfd.png)

model implement state
![image](https://user-images.githubusercontent.com/68010870/115548159-87ed0000-a27d-11eb-81a0-7772008e0d3b.png)

dev-environment:
-windows: 10
-php: 7.4.7
-composer: 2.0.2

prod-environment:
-clearOS: 7.8
-php: 7.4.14
-composer: 1.10.20

Problem: in my dev-enviroment y can access to $name attribute like this ($step === SolicitudAbiertaState::$name). But in my prod-enviroment i has this problem:
"message": "syntax error, unexpected 'Model' (T_STRING), expecting function (T_FUNCTION) or const (T_CONST)", "exception": "ParseError", "file": ".../vendor/spatie/laravel-model-states/src/State.php"

I hope u cant give a hand.
kind regards.

**Reply by brendt** [reply]
That sounds like a PHP version mismatch of some sorts, are you sure you're using PHP 7.4 on your prod env? Please provide the output of `phpversion()` within the web context (not the CLI) on your production server.

**Reply by mtamagusku** [reply]
I leave u here some server data:
![image](https://user-images.githubusercontent.com/68010870/115720433-7413cd80-a353-11eb-9cb7-48cb87f58989.png)
![image](https://user-images.githubusercontent.com/68010870/115720637-a1f91200-a353-11eb-80de-238fc083cd35.png)

Sorry but I dont manage the server, that is the info, they give me.
Thx, again.

**Reply by freekmurze** [reply]
> { "message": "syntax error, unexpected 'Model' (T_STRING), expecting function (T_FUNCTION) or const (T_CONST)", "exception": "ParseError", "file": "/var/www/virtual//html/vendor/spatie/laravel-model-states/src/State.php", "line": 17.

We guarantee that there is no syntax error in our code. The tests run perfectly. 
We won't be able to help you with this problem.

**Reply by mtamagusku** [reply]
Thx a lot any way.
